packages/hub/node-tests/cards-test.js

```diff
@@ -1,165 +1,63 @@
 
+const JSONAPIFactory = require('@cardstack/test-support/jsonapi-factory');
+const temp = require('@cardstack/test-support/temp-helper');
 const {
   createDefaultEnvironment,
   destroyDefaultEnvironment
 } = require('../../../tests/stub-searcher/node_modules/@cardstack/test-support/env');
 
 describe('hub/cards', function () {
-  let env;
+  let env, searchers, path;
+
+  beforeEach(async function () {
+    let path = await temp.mkdir('cards');
+
+    // This data source will monitor a a glob for card json:api documents.
+    // When it detects a new card document, the data source will indexing the card,
+    // generate the necessary schema, serialize templates, css, js to disk in a
+    // folder that ember-cli is watching, such that ember-cli will dynamically
+    // build and serve the new card assets
+    factory.addResource('data-sources', 'card-loader')
+      .withAttributes({
+        sourceType: '@cardstack/hub::card-loader',
+        params: {
+          // this is the glob that we watch in order to discover cards
```

Comment:
On closer inspection, this is an area we will want to change. Card discovery needs to be done by indexers and/or searchers. It's not a special code path.

Which is good because it means you can load the JSON:API documents representing cards using all the existing tools, and implementing a "card NPM package searcher" is a separable piece of work that's not blocking to any of the other work.

In other words, loading code out of any card always goes through these steps:

1. Some indexer or searcher gives you a json:api document of type card. This is a complete description of the card, no other inputs needed. 
2. When you want to incorporate the card's code into the app (either server or client code), the hub will need to convert the json:api document to actual files on disk so that 
    1. yarn install can run and ensure all dependencies are present.
    2. the card's code is present in conventionally-located files on disk that can be `require`ed by the hub or `import()`ed into the ember app.

As an optimization, it would be good to be able to avoid writing new files in step 2 when the original data source already happens to be based off local files on disk. But that is strictly an optimization and things must work when it isn't true.

Reply:
so maybe like a `hub:card-loader` service that indexers and searchers can use when they have received a card that they want to load into the index?

Reply:
I guess also, actually creating test card node modules might be a little premature--as i wouldn't need to have a module in order to write a test--simply the card document would do. (but I could at perhaps build the card document in a manner that would look like how a developer would interact with the JSONAPIFactory for creating a card, and imbue the document with the compound ID's using test helpers)

Reply:
just thinking out load, i wonder if the new wrapper API for dealing with cards and the card loaded actually live in the same service, maybe like `hub:card-services`? which would include functions for loading cards and retrieving cards

Reply:
Yeah I think that makes sense. Then even things like the retrieving a computed property implementation would go through the card service to load the code.